course_catalog/etl/youtube.py

```diff
@@ -82,6 +84,67 @@ def get_videos_for_playlist(youtube_client, playlist_id, offered_by):
             break
 
 
+def get_captions_for_video(video):
+    """
+    Function which fetches and returns xml captions for a video object
+
+    Args:
+        video (object) : Video Object
+
+    Returns:
+        a string with the transcript in xml format
+
+    """
+
+    pytube_client = pytube.YouTube(video.url)
+    all_captions = pytube_client.captions.all()
+
+    selected_capiton = next(
+        (
+            caption
+            for caption in all_captions
+            if caption.code == "en" and "auto-generated" not in caption.name
+        ),
+        None,
+    )
+
+    if not selected_capiton:
+        selected_capiton = next(
+            (caption for caption in all_captions if caption.code == "en"), None
+        )
+
+    if selected_capiton:
+        return selected_capiton.xml_captions
+
+
+def parse_video_captions(xml_captions):
+    """
+    Function which parses xml captions and returns a string with just the transcript
+
+    Args:
+        xml_captions (string) : Caption with xml with text and timestamps
+
+    Returns:
+        a string with the transcript with timestamps removed
+
+    """
+
+    if not xml_captions:
+        return ""
+
+    transcript = ""
+    root = ElementTree.fromstring(xml_captions)
+
+    for child in root.iter():
+        text = child.text or ""
+        text = unescape(text.replace("\n", " ").replace("  ", " "))
+
+        if text:
+            transcript += " " + text
```

Comment:
It's best to avoid string concatenation when building up a string - every time you have a `+` here it adds the overhead of creating a new string, so in this case the code is potentially generating 2 new strings per item. It's probably not much in the small scale but it can add up.

A more performant way to approach this is typically to append each piece of text to a list and then use [`str.join`](https://docs.python.org/2/library/string.html#string.join) to combine them in a single operation.